lib/database_cleaner/active_record/truncation.rb

```diff
@@ -27,7 +27,7 @@ def database_cleaner_view_cache
 
       def database_cleaner_table_cache
         # the adapters don't do caching (#130) but we make the assumption that the list stays the same in tests
-        @database_cleaner_tables ||= tables
+        @database_cleaner_tables ||= (::Rails::VERSION::MAJOR >= 5 ? data_sources : tables)
```

Comment:
Changing this to `::ActiveRecord::VERSION::MAJOR >= 5` should make the tests pass.